packages/wallet/src/test-utils/launchNode.test.ts

```diff
@@ -0,0 +1,21 @@
+import { Provider } from '@fuel-ts/providers';
+
+import { WalletUnlocked } from '../wallets';
+
+import { launchNodeAndGetWallets } from './launchNode';
+
+test('launchNodeAndGetWallets', async () => {
```

Comment:
Please add more tests covering the duality between `default` configs vs.`user` configs.

 - User **_not_** passing a `chainConfig`
 - User passing `useSystemFuelCore = true`
 - User not passing `walletCount`
 - User passing a different `walletCount` (other than 10)
 - Method `launchNodeAndGetWallets` being called without params

Reply:
@arboleya great points.

- I actually had a test case for `useSystemFuelCore = true` in place but it won't work in our CI :( 
- Shouldn't the last test case you mentioned cover the user not passing a chainconfig and not passing a walletcount?

Reply:
Hmm I don't think so, perhaps. Let's see.

I should've split the observations for each method, here it goes:

`launchNode()`:
 - User not passing a chainConfig
 - User passing useSystemFuelCore = true

`launchNodeAndGetWallets()`:
 - User not passing `walletCount`
 - User passing a different `walletCount` _(other than 10, the default)_
 - Method being called without any params

Makes sense?

Reply:
With relation to the `useSystemFuelCore=true` case, one alternative would be to mock the `spawn` method and assert the 1st argument to ensure it's `fuel-core`. It may be tricky to mock this one, though.

Pls, let me know if it becomes a PITA.

Reply:
@arboleya the test cases look good collectively, but I feel like we only really need to test the `launchNodeAndGetWallets` function properly since it internally always calls `launchNode` and just passes down the arguments. If there is a problem anywhere in this, it will be caught in our assertions. I was thinking we can have these three test cases + one for `useSystemFuelCore: true`

<img width="783" alt="image" src="https://github.com/FuelLabs/fuels-ts/assets/39617427/28e21725-b457-47e7-b44b-eb1e8650b8df">


Thoughts?

Reply:
Sure, these might be enough, and I also don't want to hold this back.

If the `useSystemFuelCore` test becomes difficult, let's go without it.

Just make **_sure_** it's working with your local `fuel-core`, and voilà.

You can also check for possible missed spots by checking the coverage report.

```shell
# tab1
pnpm node:run

# tab 2
pnpm test packages/wallet/src/test-utils/launchNode.test.ts
open ./coverage/lcov-report/index.html
```

Reply:
Never mind, you already did it.

I missed the last commits when sending the previous comment.